app/gui/src/controller/searcher/component.rs

```diff
@@ -171,21 +177,46 @@ impl Component {
         });
 
-        // Pick the best match score of the two, use only the character indices matching the label.
-        let subsequence = match (label_subsequence, code_subsequence) {
-            (Some(label), Some(code)) => {
-                let score = label.score.max(code.score);
-                Some(fuzzly::Subsequence { score, ..label })
-            }
-            (None, Some(code)) => Some(fuzzly::Subsequence { indices: Vec::new(), ..code }),
-            (Some(label), None) => Some(label),
+        // Pick the best match between label and code matches. Use only the character indices of the
+        // label match, or an empty array of indices if there is no label match.
+        let label_or_code_match = match (label_subsequence, code_subsequence) {
+            (Some(label), None) => Some((label, MatchKind::Label)),
+            (Some(label), Some(code)) if label.score >= code.score =>
+                Some((label, MatchKind::Label)),
+            (Some(label), Some(code)) =>
+                Some((fuzzly::Subsequence { indices: label.indices, ..code }, MatchKind::Code)),
+            (None, Some(code)) =>
+                Some((fuzzly::Subsequence { indices: Vec::new(), ..code }, MatchKind::Code)),
             (None, None) => None,
         };
 
-        *self.match_info.borrow_mut() = match subsequence {
-            Some(subsequence) => MatchInfo::Matches { subsequence },
-            None => MatchInfo::DoesNotMatch,
+        // Match the input pattern to an entry's aliases, if available, and select the best match.
+        let alias_match = self.aliases().map(|aliases| {
+            let alias_match = aliases.filter_map(|alias| {
+                if fuzzly::matches(alias, pattern.as_ref()) {
+                    let metric = fuzzly::metric::default();
+                    let subsequence =
+                        fuzzly::find_best_subsequence(alias, pattern.as_ref(), metric);
+                    subsequence.map(|subsequence| (subsequence, alias))
+                } else {
+                    None
+                }
+            });
+            alias_match.max_by(|(lhs, _), (rhs, _)| lhs.compare_scores(rhs))
+        });
+        let alias_match = alias_match.flatten();
+
+        // Pick the best match between the previous label or code match and the alias match.
+        let match_info = match (label_or_code_match, alias_match) {
+            (Some((subsequence, kind)), None) => MatchInfo::Matches { subsequence, kind },
+            (Some((subsequence, kind)), Some((alternative, _)))
+                if subsequence.score >= alternative.score =>
+                MatchInfo::Matches { subsequence, kind },
+            (_, Some((subsequence, alias))) =>
+                MatchInfo::Matches { subsequence, kind: MatchKind::Alias(alias.to_im_string()) },
+            (None, None) => MatchInfo::DoesNotMatch,
         };
+        *self.match_info.borrow_mut() = match_info;
```

Comment:
This section became a bit complicated.

I think it would be more readable to make a single iterator over `Option<MatchInfo>`, created from a chain of label match, code match and aliases match, and then we could take max_by from this single iterator.